lib/cell.rb

```diff
@@ -28,15 +27,55 @@ def fire_upon
     if @empty == false
      @ship.hit
      @render = "H"
-    else
-     @render = "M"
+    # else
+    #  @render = "M"
     end
   end
 
   def render(reveal = false)
-    if reveal == true
-      @render = "S"
-    else @render
+    if @ship
```

Comment:
Always consider every possibility when using a optional argument for a method. Deeply nested conditionals are possible but not ideal (not clean)